pkg/scheduler/factory/factory.go

```diff
@@ -483,21 +511,53 @@ func (c *Configurator) getPriorityMetadataProducer() (priorities.PriorityMetadat
 	return getPriorityMetadataProducer(*pluginArgs)
 }
 
-// GetPredicateMetadataProducer returns a function to build Predicate Metadata.
+// getPredicateMetadataProducer returns a function to build Predicate Metadata.
 // It is used by the scheduler and other components, such as k8s.io/autoscaler/cluster-autoscaler.
 func (c *Configurator) GetPredicateMetadataProducer() (predicates.PredicateMetadataProducer, error) {
 	return getPredicateMetadataProducer()
 }
 
-// GetPredicates returns the predicate functions.
-// It is used by the scheduler and other components, such as k8s.io/autoscaler/cluster-autoscaler.
-func (c *Configurator) GetPredicates(predicateKeys sets.String) (map[string]predicates.FitPredicate, error) {
+func (c *Configurator) getPredicateConfigs(predicateKeys sets.String) (map[string]predicates.FitPredicate, *config.Plugins, []config.PluginConfig, error) {
+	predicateConfigs := make(map[string]predicates.FitPredicate)
 	pluginArgs, err := c.getPluginArgs()
 	if err != nil {
-		return nil, err
+		return nil, nil, nil, err
 	}
 
-	return getFitPredicateFunctions(predicateKeys, *pluginArgs)
+	initialConfigs, err := getFitPredicateFunctions(predicateKeys, *pluginArgs)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	asPlugins := sets.NewString()
+	for predicateKey := range initialConfigs {
+		if _, exist := predicateToFrameworkConfigProducer[predicateKey]; !exist {
+			predicateConfigs[predicateKey] = initialConfigs[predicateKey]
+			continue
+		}
+		asPlugins.Insert(predicateKey)
+	}
+
+	// Append the plugins according to the predicates specified order.
+	var plugins config.Plugins
+	var pluginConfig []config.PluginConfig
+	for _, predicateKey := range predicates.Ordering() {
+		if asPlugins.Has(predicateKey) {
+			producer := predicateToFrameworkConfigProducer[predicateKey]
+			p, pc := producer()
+			plugins.Append(&p)
+			pluginConfig = append(pluginConfig, pc...)
+			asPlugins.Delete(predicateKey)
+		}
+	}
+
+	for predicateKey := range asPlugins {
```

Comment:
So predicateKeys not existing in predicate.Ordering() is added after. I think a comment should be added here to highlight this.

Reply:
Done.